Index: docroot/sites/all/modules/contrib/pdf_to_imagefield/pdf_to_image.module

===================================================================
--- docroot/sites/all/modules/contrib/pdf_to_imagefield/pdf_to_image.module	(date 1414061219000)
+++ docroot/sites/all/modules/contrib/pdf_to_imagefield/pdf_to_image.module	(date 1414675248000)
@@ -427,6 +427,7 @@
       'entity' => $entity, // Don't actually need the whole thing, just the id really.
       'entity_type' => $entity_type,
       'entity_id' => $id,
+      'entity_vid' => $vid,
       'image'  => array(
         'field'    => field_info_field($target_field),
         'instance' => field_info_instance($entity_type, $target_field, $field_instance['bundle']),
@@ -568,12 +569,13 @@
 
   $field_name = $results['params']['image']['field']['field_name'];
   $entity_id = $results['params']['entity_id'];
+  $entity_vid = $results['params']['entity_vid'];
   $entity_type = $results['params']['entity_type'];
 
   if (isset($entity_id) && is_numeric($entity_id)) {
     // Don't use the entity as given, load it again as things may have happened
     // to it since the batch job began.
-    $entity = entity_load_unchanged($entity_type, $entity_id);
+    $entity = entity_revision_load($entity_type, $entity_vid);
     if (is_object($entity)) {
       //Set translation as an empty array and reset static cache
       //This forces field_language to return the correct lang_code
