diff --git a/docroot/sites/all/modules/contrib/uuid_features/includes/uuid_node.features.inc b/docroot/sites/all/modules/contrib/uuid_features/includes/uuid_node.features.inc
index 54974d2..abdbfbc 100755
--- a/docroot/sites/all/modules/contrib/uuid_features/includes/uuid_node.features.inc
+++ b/docroot/sites/all/modules/contrib/uuid_features/includes/uuid_node.features.inc
@@ -184,6 +184,19 @@ function uuid_node_features_rebuild($module) {
         }
       }
 
+      // Workbench integration.
+      if (!empty($node->workbench_moderation)) {
+        if (!empty($existing->workbench_moderation)) {
+          $node->workbench_moderation = $existing->workbench_moderation;
+        }
+        else {
+          if (!empty($node->workbench_moderation['published']['published'])) {
+            $node->workbench_moderation_state_new = 'published';
+          }
+          unset($node->workbench_moderation);
+        }
+      }
+
       // The hook_alter signature is:
       // hook_uuid_node_features_rebuild_alter(&node, $module);
       drupal_alter('uuid_node_features_rebuild', $node, $module);
