diff --git a/includes/features.menu.inc b/includes/features.menu.inc
index c883e6a..9129e27 100644
--- a/includes/features.menu.inc
+++ b/includes/features.menu.inc
@@ -224,6 +224,20 @@ function menu_links_features_export_render($module, $data, $export = NULL) {
     if ($link = features_menu_link_load($identifier)) {
       $new_identifier = menu_links_features_identifier($link, empty($export));

+      // Export entity UUID.
+      if (isset($link['router_path'])) {
+        switch ($link['router_path']) {
+          case 'node/%':
+            $nid = str_replace('node/', '', $link['link_path']);
+            $item_uuid = db_select('node', 'n')
+              ->fields('n', array('uuid'))
+              ->condition('nid', $nid)
+              ->execute()
+              ->fetchField();
+            $link['item_uuid'] = $item_uuid;
+            break;
+        }
+      }
       // Replace plid with a parent path.
       if (!empty($link['plid']) && $parent = menu_link_load($link['plid'])) {
         // If the new identifier is different than the old, maintain
@@ -341,6 +355,25 @@ function menu_links_features_rebuild_ordered($menu_links, $reset = FALSE) {
       else {
         $link['plid'] = 0;
       }
+
+      // Set the path to use entity id loaded with UUID.
+      if (!empty($link['item_uuid']) && !empty($link['router_path'])) {
+        switch ($link['router_path']) {
+          case 'node/%':
+            $entity_id = db_select('node', 'n')
+              ->fields('n', array('nid'))
+              ->condition('uuid', $link['item_uuid'])
+              ->execute()
+              ->fetchField();
+            if (!empty($entity_id)) {
+              $link['link_path'] = str_replace('%', $entity_id, $link['router_path']);
+            }
+            else {
+              $link['link_path'] = '<front>';
+            }
+            break;
+        }
+      }
       menu_link_save($link);
     }
   }
