diff --git a/docroot/sites/all/modules/contrib/link/link.module b/docroot/sites/all/modules/contrib/link/link.module
index b0c53c3..47bf3e7 100644
--- a/docroot/sites/all/modules/contrib/link/link.module
+++ b/docroot/sites/all/modules/contrib/link/link.module
@@ -647,11 +647,12 @@ function _link_parse_url($url) {
     $url = substr($url, 0, strpos($url, '#'));
   }
   // Separate out the query string, if any.
-  if (strpos($url, '?') !== FALSE) {
-    $query = substr($url, strpos($url, '?') + 1);
-    $url_parts['query'] = _link_parse_str($query);
-    $url = substr($url, 0, strpos($url, '?'));
-  }
+  // see issues: https://www.drupal.org/node/1984398 , https://www.drupal.org/node/1646360
+//  if (strpos($url, '?') !== FALSE) {
+//    $query = substr($url, strpos($url, '?') + 1);
+//    $url_parts['query'] = _link_parse_str($query);
+//    $url = substr($url, 0, strpos($url, '?'));
+//  }
   $url_parts['url'] = $url;
   return $url_parts;
 }
